sources/packages/forms/src/form-definitions/aestaddnewsin.json

```diff
@@ -0,0 +1,541 @@
+      "protected": false,
+      "unique": false,
+      "persistent": false,
+      "clearOnHide": true,
+      "refreshOn": "",
+      "redrawOn": "",
+      "tableView": false,
+      "dataGridLabel": false,
+      "labelPosition": "top",
+      "description": "",
+      "errorLabel": "",
+      "tooltip": "",
+      "hideLabel": false,
+      "tabindex": "",
+      "disabled": false,
+      "autofocus": false,
+      "dbIndex": false,
+      "customDefaultValue": "",
+      "calculateValue": "",
+      "calculateServer": false,
+      "widget": null,
+      "validateOn": "change",
+      "validate": {
+        "required": false,
+        "custom": "",
+        "customPrivate": false,
+        "strictDateValidation": false,
+        "multiple": false,
+        "unique": false
+      },
+      "allowCalculateOverride": false,
+      "encrypted": false,
+      "showCharCount": false,
+      "showWordCount": false,
+      "allowMultipleMasks": false,
+      "addons": [],
+      "id": "ejgzuz"
+    },
+    {
+      "label": "Social Insurance Number (SIN)",
+      "labelPosition": "top",
+      "labelWidth": "",
+      "labelMargin": "",
+      "placeholder": "",
+      "description": "",
+      "tooltip": "",
+      "prefix": "",
+      "suffix": "",
+      "widget": { "type": "input" },
+      "inputMask": "999 999 999",
+      "displayMask": "",
+      "allowMultipleMasks": false,
+      "customClass": "",
+      "tabindex": "",
+      "autocomplete": "",
+      "hidden": false,
+      "hideLabel": false,
+      "showWordCount": false,
+      "showCharCount": false,
+      "mask": false,
+      "autofocus": false,
+      "spellcheck": true,
+      "disabled": false,
+      "tableView": true,
+      "modalEdit": false,
+      "multiple": false,
+      "persistent": true,
+      "inputFormat": "plain",
+      "protected": false,
+      "dbIndex": false,
+      "case": "",
+      "truncateMultipleSpaces": false,
+      "encrypted": false,
+      "redrawOn": "",
+      "clearOnHide": true,
+      "customDefaultValue": "",
+      "calculateValue": "",
+      "calculateServer": false,
+      "allowCalculateOverride": false,
+      "validateOn": "blur",
+      "validate": {
+        "required": true,
+        "minLength": 11,
+        "maxLength": 11,
+        "minWords": "",
+        "maxWords": "",
+        "pattern": "",
+        "customMessage": "Invalid Social Insurance Number",
+        "custom": "valid = false;\r\nif (input) {\r\n  input = input.replace(/\\s/g, \"\");\r\n  if (input.length === 9) {\r\n    var checksum = 0;\r\n    for (var i = 0; i < input.length; i++) {\r\n      const currentDigit = parseInt(input.charAt(i));\r\n      if ((i + 1) % 2 === 0) {\r\n        const digitTimes2 = currentDigit * 2;\r\n        checksum += digitTimes2 < 10 ? digitTimes2 : digitTimes2 - 9;\r\n      } else {\r\n        checksum += parseInt(input.charAt(i));\r\n      }\r\n    } \r\n    if (checksum % 10 === 0) {\r\n      valid = true;\r\n    }\r\n  }\r\n}",
```

Comment:
Adding a comment to this validation will be helpful.

Reply:
This is the SIN common validation algorithm used all around applications to do the SIN validation.
What comment is expected here? I can add the algorithm name if it helps (e.g. _Luhn algorithm_),

Reply:
It is changed as above.

Reply:
I wasn't aware that SIN validation follows Luhn algorithm.